python/protobuf_rpc/base_channel.py

```diff
@@ -26,9 +31,17 @@ def check_for_errors(self, resp_obj):
         error_message = getattr(resp_obj, "error_message", "RPC Error")
         raise error_class(error_message)
 
-    def create_rpc_request(self, method, request):
+    def create_rpc_request(self, method, request, controller):
         rpcRequest = Request()
+        if controller.headers:
+            rpcRequest.headers.CopyFrom(controller.headers)
         rpcRequest.request_proto = request.SerializeToString()
         rpcRequest.service_name = method.containing_service.full_name
         rpcRequest.method_name = method.name
+        rpcRequest.headers.timestamp = int(round(time.time() * 1000))
+        rpcRequest.headers.hostname = socket.gethostname()
+        pid = os.getpid()
+        process_name = psutil.Process(pid)
+        rpcRequest.headers.procname = process_name
```

Comment:
Also, lets move these (hostname, pid, process_name) to be module-level variables.